kitsune/wiki/managers.py

```diff
@@ -12,26 +12,77 @@ class VisibilityManager(models.Manager):
     # should be "document".
     document_relation: str | None = None
 
+    @property
+    def document_relation_prefix(self):
+        """Provides ORM prefix for the document relation."""
+        return f"{self.document_relation}__" if self.document_relation else ""
+
     def get_creator_condition(self, user):
         """
         Return a conditional (e.g., a Q or Exists object) that is only true
         when the given user is the creator of this document or revision.
         """
         raise NotImplementedError
 
-    def visible(self, user, **kwargs):
+    def unrestricted(self, user=None, **kwargs):
+        """
+        Documents are unrestricted for a given user if the user is a superuser, or
+        if they're not restricted to staff or a specific group, or when the user is
+        staff and they're restricted by staff, or when the user is a member of the
+        specific group to which they're restricted. A translation (i.e., a document
```

Comment:
Regardless of the restriction staff members should always have visibility to restricted documents. I haven't looked at that part of the code through my review yet but by this wording I get the impression that a staff member will not have access to the document if it's restricted to a group by a non staff member.

Reply:
Yes, you're right on. I changed this in both the `unrestricted` method as well as the `visible` method.